Description: Turn type ahead find back on for the file chooser. This was disabled upstream before a replacement is ready.
Author: Iain Lane <iain.lane@canonical.com>
Bug-Ubuntu: http://pad.lv/1464654
Forwarded: not-needed

Index: gtk+3.0-3.22.29-2ubuntu1/gtk/ui/gtkfilechooserwidget.ui
===================================================================
--- gtk+3.0-3.22.29-2ubuntu1.orig/gtk/ui/gtkfilechooserwidget.ui
+++ gtk+3.0-3.22.29-2ubuntu1/gtk/ui/gtkfilechooserwidget.ui
@@ -162,7 +162,7 @@
                                   <object class="GtkTreeView" id="browse_files_tree_view">
                                     <property name="visible">1</property>
                                     <property name="has-tooltip">1</property>
-                                    <property name="enable-search">0</property>
+                                    <property name="enable-search">1</property>
                                     <child internal-child="accessible">
                                       <object class="AtkObject" id="browse_files_tree_view-atkobject">
                                         <property name="AtkObject::accessible-name" translatable="yes">Files</property>
Index: gtk+3.0-3.22.29-2ubuntu1/gtk/gtkfilechooserwidget.c
===================================================================
--- gtk+3.0-3.22.29-2ubuntu1.orig/gtk/gtkfilechooserwidget.c
+++ gtk+3.0-3.22.29-2ubuntu1/gtk/gtkfilechooserwidget.c
@@ -2427,12 +2427,12 @@
 {
   GtkFileChooserWidgetPrivate *priv = impl->priv;
 
-  gtk_tree_view_set_search_column (GTK_TREE_VIEW (priv->browse_files_tree_view), -1);
-
   gtk_tree_view_column_set_sort_column_id (priv->list_name_column, MODEL_COL_NAME);
   gtk_tree_view_column_set_sort_column_id (priv->list_time_column, MODEL_COL_TIME);
   gtk_tree_view_column_set_sort_column_id (priv->list_size_column, MODEL_COL_SIZE);
   gtk_tree_view_column_set_sort_column_id (priv->list_location_column, MODEL_COL_LOCATION_TEXT);
+
+  gtk_tree_view_set_search_column (GTK_TREE_VIEW (priv->browse_files_tree_view), MODEL_COL_NAME);
 }
 
 static gboolean
@@ -3082,6 +3082,7 @@
       search_stop_searching (impl, TRUE);
       search_clear_model (impl, TRUE);
       gtk_widget_hide (impl->priv->remote_warning_bar);
+      gtk_tree_view_set_enable_search (GTK_TREE_VIEW (impl->priv->browse_files_tree_view), TRUE);
     }
 }
 
@@ -3134,6 +3135,7 @@
   location_bar_update (impl);
   search_setup_widgets (impl);
   gtk_widget_set_sensitive (priv->filter_combo, FALSE);
+  gtk_tree_view_set_enable_search (GTK_TREE_VIEW (priv->browse_files_tree_view), FALSE);
 }
 
 static void
